src/k8s-extension/azext_k8s_extension/partner_extensions/Dapr.py

```diff
@@ -4,43 +4,143 @@
+            "Dapr installation, you must provide them in the configuration settings. Failing to do so will result in"\
+            "an error, since Helm upgrade will try to modify the StatefulSet."\
+            f"Please refer to {self.TSG_LINK} for more information."
+
+        self.RELEASE_INFO_HELP_STRING = "The Helm release name and namespace can be found by running 'helm list -A'."
+
+        # constants for error messages.
+        self.ERR_MSG_INVALID_SCOPE_TPL = "Invalid scope '{}'. This extension can be installed only at 'cluster' scope."\
+            f" Check {self.TSG_LINK} for more information."
+
+    def _get_release_info(self, release_name: str, release_namespace: str, configuration_settings: dict)\
+            -> Tuple[str, str, bool]:
+        '''
+        Check if Dapr is already installed in the cluster and get the release name and namespace.
+        If user has provided the release name and namespace in configuration settings, use those.
+        Otherwise, prompt the user for the release name and namespace.
+        If Dapr is not installed, return the default release name and namespace.
+        '''
+        name, namespace, dapr_exists = release_name, release_namespace, False
+
+        # Set the default release name and namespace if not provided.
+        name = name or self.DEFAULT_RELEASE_NAME
+        namespace = namespace or self.DEFAULT_RELEASE_NAMESPACE
+
+        if configuration_settings.get(self.SKIP_EXISTING_DAPR_CHECK_KEY, 'false') == 'true':
+            logger.info("%s is set to true, skipping existing Dapr check.", self.SKIP_EXISTING_DAPR_CHECK_KEY)
+            return name, namespace, False
+
+        cfg_name = configuration_settings.get(self.EXISTING_DAPR_RELEASE_NAME_KEY, None)
+        cfg_namespace = configuration_settings.get(self.EXISTING_DAPR_RELEASE_NAMESPACE_KEY, None)
 
-    def Create(self, cmd, client, resource_group_name, cluster_name, name, cluster_type, cluster_rp,
-               extension_type, scope, auto_upgrade_minor_version, release_train, version, target_namespace,
-               release_namespace, configuration_settings, configuration_protected_settings,
-               configuration_settings_file, configuration_protected_settings_file):
+        # If the user has specified the release name and namespace in configuration settings, then use it.
+        if cfg_name and cfg_namespace:
+            logger.info("Using the release name and namespace specified in the configuration settings.")
+            return cfg_name, cfg_namespace, True
 
+        # If either release name or namespace is missing, ignore the configuration settings and prompt the user.
+        if cfg_name or cfg_namespace:
+            logger.warning("Both '%s' and '%s' must be specified in the configuration settings. Only one of them is "
+                           "specified, ignoring.", self.EXISTING_DAPR_RELEASE_NAME_KEY,
+                           self.EXISTING_DAPR_RELEASE_NAMESPACE_KEY)
```

Comment:
Your are defaulting to existing instance of Dapr not existing (line 64) .  So, how are you handling the case one of these are not passed in?  As is, you will be returning false i.e. it doesn't exist and proceed with installation, which would fail, no?

Reply:
If user does not pass a flag, we explicitly prompt and ask them.